packages/core/src/state-management/anchor-resuming-service.ts

```diff
@@ -33,24 +37,37 @@ export class AnchorResumingService {
     })
   }
 
+  delay(ms: number): Promise<void> {
+    return new Promise((resolve) => setTimeout(resolve, ms))
+  }
+
   async resumeRunningStatesFromAnchorRequestStore(repository: Repository): Promise<void> {
     if (this.#shouldBeClosed) {
       throw Error('This AnchorResumingService is closed, create a new instance to resume')
     }
 
+    this.logger.imp(`Resuming polling for streams with pending anchors`)
+
+    let numRestoredStreams = 0
     let gt: StreamID | undefined = undefined
     let batch = new Array<AnchorRequestStoreListResult>()
     do {
       batch = await repository.anchorRequestStore.list(RESUME_BATCH_SIZE, gt)
-      batch.forEach((listResult) => {
+      for (const listResult of batch) {
         this.resumeQ.add(async () => {
           await repository.fromMemoryOrStore(listResult.key)
           this.logger.verbose(`Resumed running state for stream id: ${listResult.key.toString()}`)
+          numRestoredStreams++
         })
-      })
+        await this.delay(RATE_LIMIT_DELAY)
```

Comment:
We probably don't need to hardcode this to 100, but rather jitter it between like 10 and 50ms.

Reply:
done